presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -317,6 +327,13 @@ public void testOrderByMustAppearInSelectWithDistinct()
     }
 
     @Test
+    public void testNonDeterministicOrderBy()
+    {
+        analyze("SELECT DISTINCT random() as b FROM t1 ORDER BY b");
+        assertFails(NONDETERMINISTIC_ORDER_BY_EXPRESSION, "SELECT DISTINCT random() FROM t1 ORDER BY random()");
```

Comment:
* add a test for `SELECT a FROM t1 ORDER BY random()`
* in the failing example is the fact you used `random()` in DISTINCT important? Should the query fail the same, if it were `SELECT DISTINCT a ...` ? Maybe you should test both cases?